### ISSUE TITLE
WeTek Video stuttering while playing 24p videos

### ISSUE THREAD

**Issue by suchmich1983** [OpenELEC/OpenELEC.tv#4259]
I do notice stuttering while playing 24p video content on my WeTek.Play. There are 1-2 "micro-stutters" per minute. It doesnt matter what content (tried a lot of files) or where the content is coming from (local, NFS Share, Internet...), its always stuttering..

Please find a few logs attached:
Debug Logs: http://sprunge.us/KXWJ  and http://sprunge.us/SeBL

This is an example file I get from one user in OE Support Chat. "https://dl.dropboxusercontent.com/u/55728161/clannad.mkv" One played via 1080p and "change refresh rate to match video" and one was played at 720p60Hz.

Dmesg Log 5 seconds after the Video was started:
http://sprunge.us/GUSa

And here are Media-Infos about the 2 videos that were played during test:
http://pastebin.com/mVhgfCma
http://pastebin.com/9RU20ehP

here is a little part of the chat I had within the OE Support Chat regarding this issue:

[14:42] fritsch okay - so - what are we doing here, as I am not the guy to talk away your bugs :-) cause I see the issue
[14:42] fritsch we need:
[14:43] fritsch a) a video mode that is properly announced (23.976)
[14:43] fritsch so that kodi knows where to sync to, as itself thinks it uses 24.0
[14:43] fritsch b) we need a proper video clock, so that people with a 24.0hz only TV can watch 23.976 content with audio resampling
[14:43] fritsch the video clock would tick 24.0 times a second and therefore audio needs to be speed up a bit
[14:44] fritsch are you able to hack on the kernel?
[14:44] suchmich1983 I have some little LINUX knowledge, but I do not know how by myself
[14:45] fritsch oki
[14:45] fritsch so - in fact: it won't get better as is now
[14:45] fritsch sadly
[14:45] fritsch i also cannot promise anything, as i am just a private person hacking for fun
[14:45] fritsch but this issue is solvable!
[14:45] suchmich1983 but when I understand you correctly, it could get better in the future?
[14:45] fritsch the hw is capable of doing those things
[14:46] fritsch yeah
[14:46] suchmich1983 ok, sound good for me

**Comment by r2ixuz** [comment]
I can confirm such a micro stutter for OE 5.95.5 on wetek play. It looks like every 2-4 minutes some frames get lost, so you can see barely a little stutter.

**Comment by r2ixuz** [comment]
For some more clarification have a look at this post: http://wetekforums.com/vb5/forum/wetek-play/linux/openelec/16667-video-stutters-on-openelec-5-95-5?p=17819#post17819

**Comment by jelbo** [comment]
Can confirm on WeTek OpenELEC Box. Here are some test videos: https://onedrive.live.com/redir?resid=700C1174B49FE3CE!6810&authkey=!ABi46eYO2UFHMWI&ithint=folder%2cmkv

**Comment by grin1** [comment]
It is so frustrating to not become a statement of an OE developer. Come one, you sell the WeTek Play as an official device for watching movies using OE and it is barely unusable. How can we enjoy watching movies with that micro stutters? There are several threads of people complaining in the wetekforums and OEforums. There are several debug logs, there are several test videos. The problem is well known but there is not a single statement of a developer. Yes, we know that all OE developers work in there spare time and that it is a open source project. But as soon as people pay money for a device they expect it to work. All we want is a statement that tells us how the status of the whole problem is, what will be done and when we can expect a fix of this...

**Comment by grin1** [comment]
It is so frustrating to not become a statement of an OE developer. Come one, you sell the WeTek Play as an official device for watching movies using OE and it is barely unusable. How can we enjoy watching movies with that micro stutters? There are several threads of people complaining in the wetekforums and OEforums. There are several debug logs, there are several test videos. We have done our homework. The problem is well known but there is not a single statement of a developer. Yes, we know that all OE developers work in there spare time and that it is a open source project. But as soon as people pay money for an official device they expect it to work. All we want is a statement that tells us how the status of the whole problem is, what will be done and when we can expect a fix of this... Remember: The problem came up in February/March, almost a year ago.

Ping @codesnake @sraue @fritsch

**Comment by codesnake** [comment]
@grin1 Don't worry. I'm working on this issue. Maybe you'll feel better now :) But in my own opinion, you are exaggerating the problem too much. I use WeTek Play everyday and the issue is not noticeable in most cases. But you state that device is "unusable" and doesn't work.

**Comment by fritsch** [comment]
@grin1 I don't work on WT Play anymore ... but I read all your posts anyways.

**Comment by suchmich1983** [comment]
@codesnake 
do you have any updates for us regarding this issue? Are you still working on this issue?
Thx

**Comment by codesnake** [comment]
@suchmich1983 Still working. But need to do more priority things first.

**Comment by chewitt** [comment]
@codesnake - can we have an update please - fixes need to be done for OE 7.0

**Comment by suchmich1983** [comment]
@codesnake any news since last report?

**Comment by codesnake** [comment]
Unfortunately currently I cannot find a way to fix the issue.

**Comment by suchmich1983** [comment]
Well that was what I have expected. Did someone know who wrote the fix for the Raspi in the past?
Ping @fritsch @popcornmix @chewitt

**Comment by fritsch** [comment]
@suchmich1983 https://github.com/OpenELEC/OpenELEC.tv/issues/4259#issuecomment-154808048

**Comment by popcornmix** [comment]
I don't believe any Pi fixes will be relevant. This sounds like a hardware/firmware/kernel issue specific to Wetek (but I don't know much about that platform).

**Comment by suchmich1983** [comment]
@fritsch that was not my question :)

@popcornmix 
Wasnt there a similar development for the Pi in the past? So when I remember correctly, that Hardware wasnt able to play 23,976 Material "stutterfree" from the beginning. Or am I wrong?
Cause WeTek is also ARM based, I thought the fix could be found in the same direction....

**Comment by chewitt** [comment]
@suchmich1983 - arm CPU but different GPU's, which is the essential point you're missing

**Comment by suchmich1983** [comment]
ok, thanks for pointing me in the right direction. 
So hopefully someone will find a way to fix that issue. For Home-Cinema Enthusiasts like me, it is really annoying and as a promised "perfect media player" unacceptable..

**Comment by codesnake** [comment]
@suchmich1983 I'm still trying to find a way to fix the issue. Once I have a solution I'll PR it here.

**Comment by suchmich1983** [comment]
@codesnake 
Do you have any update? Is the search still ongoing?  Do you have an idea what is causing the issue and maybe howto fix it? Would be nice, if you can send us an update regarding this issue!
Thanks

**Comment by codesnake** [comment]
https://github.com/xbmc/xbmc/pull/9152#issuecomment-185395897

**Comment by suchmich1983** [comment]
ok, still no progress as I can see. Did the temp fix helped?
Something I can help? Maybe test or so?

**Comment by lama0900** [comment]
@suchmich1983 

unfortunately this "temporary fix" is no fix at all, because it only brings aml decoder back to life again.
before this "temporary fix" it didn't work. 

so we'll have to wait for @codesnake s new decoder.

**Comment by suchmich1983** [comment]
Its been a while since my last request. @codesnake do you have an update for us? Is the work still ongoing?
Thanks and best Regards
Christian

**Comment by lama0900** [comment]
#https://github.com/xbmc/xbmc/pull/9432

**Comment by suchmich1983** [comment]
Well... I see... thank you @lama0900 for pointing me to this direction. 
I do own my WeTek now for more than 1,5 years. It is really annoying that this 24p issue is not solvable. The WeTek could really be a perfect device but it seems that it was a 99€ waste of money...

Thanks and have a good Day..
Christian

**Comment by lama0900** [comment]
Same for me, I have mine nearly 1 yr. But Kodi v17 is alpha state and aml hw decoding is broken.
So they should fix it before v17 arrives beta/rc state, I think. 

Best Regards

**Comment by suchmich1983** [comment]
@codesnake just wanted to here if there are any news regarding this?
thx

**Comment by fritsch** [comment]
Try with latest v17 images.

**Comment by suchmich1983** [comment]
@fritsch cant find a kodi v17 image for WeTek Play. Do you have a Download-Link?

**Comment by fritsch** [comment]
No.

2016-06-26 12:07 GMT+02:00 suchmich1983 <notifications@github.com>:

> @fritsch <https://github.com/fritsch> cant find a kodi v17 image for
> WeTek Play. Do you have a Download-Link?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenELEC/OpenELEC.tv/issues/4259#issuecomment-228593691>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABCfHRN1UsPdfIr_0iUY6My3fDiLs6WKks5qPk9YgaJpZM4FjfJa>
> .
>



-- 
                   Key-ID:     0x1A995A9B
                   keyserver: pgp.mit.edu
==============================================================
Fingerprint: 4606 DA19 EC2E 9A0B 0157  C81B DA07 CF63 1A99 5A9B

**Comment by suchmich1983** [comment]
Are there any changes in v17 regarding this 24p issue? Or is it just an idea?
Anyone else does have a build to download? Or already tested it?

**Comment by lama0900** [comment]
look at your notifications at wetekforums @suchmich1983 
there you will find all the info about the link to dev build and some report from me. github is not the best place for discussing this, I think. but we've to wait for v17 progress. there are some changes, at least to the decoder, not specific to this bug.
